src/transport/mqtt/crypto.rs

```diff
@@ -78,7 +71,7 @@ impl Bundle {
         dn.push(DnType::CommonName, format!("{}/{}", realm, device_id));
 
         // Generate a random private key
-        let key_pair = KeyPair::generate_for(&PKCS_ECDSA_P384_SHA384)?;
+        let key_pair = KeyPair::generate_for(&PKCS_ECDSA_P256_SHA256)?;
```

Comment:
it is not a problem, but why do you decrement curve and SHA value to `256`? Did you see a performance problem?

Reply:
I changed it while tasting the self signing of the CSR

https://github.com/astarte-platform/astarte-device-sdk-rust/blob/0818d353bd45a349c0fbe7bb28bb50bc346fb817/src/transport/mqtt/pairing.rs#L292-L300

The validation of the CSR was failing, so I wanted a more common certificate. In the end it was the OpenSSL one that was really particular and not supported by rcgen/rustls (for the self signing and validation of the CSR). So I end up removing OpenSSL, since it wasn't actually used in the final binaries.

I think we can use the PKCS_ECDSA_P384_SHA384 private key type without problems. Still  we can consider moving to the PKCS_ECDSA_P256_SHA256, since it's the one suggested by `rcgen/rustls` and we don't have a reason to use the PKCS_ECDSA_P384_SHA384.

Reply:
The `self_sign_csr_to_pem` uses the key generated by `astarte-cfssl-config`, and it's the PKCS_ECDSA_P256_SHA256 one.

Reply:
The old certificate would still work, and I tested:

- running the e2e-tests on release-0.9, which generates the old cert and private key in the store directory
- Move to the PR branch and re-launch the e2e-tests using the same certificate and key without a problem

The new one will be re-generate when the old one expires or the store-directory is cleaned, or the device is factory reset